modules/hivemq-edge-module-etherip/src/main/java/com/hivemq/edge/adapters/etherip/EipProtocolAdapterFactory.java

```diff
@@ -90,28 +104,54 @@ public EipProtocolAdapterFactory(final boolean writingEnabled) {
 
     private static @NotNull EipAdapterConfig tryConvertLegacyConfig(
             final @NotNull ObjectMapper objectMapper,
-            final @NotNull Map<String, Object> config) {
+            final @NotNull Map<String, Object> config,
+            final @NotNull ProtocolAdapterTagService tagService) {
         final LegacyEipAdapterConfig legacyEipAdapterConfig =
                 objectMapper.convertValue(config, LegacyEipAdapterConfig.class);
 
-        final List<EipToMqttMapping> eipToMqttMappings = legacyEipAdapterConfig.getSubscriptions()
-                .stream()
-                .map(context -> new EipToMqttMapping(context.getDestinationMqttTopic(),
-                        context.getQos(),
-                        context.getMessageHandlingOptions(),
-                        context.getIncludeTimestamp(),
-                        context.getIncludeTagNames(),
-                        context.getTagName(),
-                        context.getTagAddress(),
-                        context.getDataType(),
-                        context.getUserProperties())).collect(Collectors.toList());
-
-        final EipToMqttConfig eipToMqttConfig =
-                new EipToMqttConfig(legacyEipAdapterConfig.getPollingIntervalMillis(),
-                        legacyEipAdapterConfig.getMaxPollingErrorsBeforeRemoval(),
-                        legacyEipAdapterConfig.getPublishChangedDataOnly(),
-                        eipToMqttMappings);
+        // reference tag in the config
+
+        final List<EipToMqttMapping> eipToMqttMappings = new ArrayList<>();
+        for (final LegacyEipAdapterConfig.PollingContextImpl context : legacyEipAdapterConfig.getSubscriptions()) {
+            // create tag first
+            final ProtocolAdapterTagService.AddStatus addStatus = tagService.addTag(legacyEipAdapterConfig.getId(),
+                    PROTOCOL_ID, new EipTag(context.getTagName(), new EipTagDefinition(context.getTagAddress())));
+            switch (addStatus) {
+                case SUCCESS:
+                    eipToMqttMappings.add(new EipToMqttMapping(context.getDestinationMqttTopic(),
+                            context.getQos(),
+                            context.getMessageHandlingOptions(),
+                            context.getIncludeTimestamp(),
+                            context.getIncludeTagNames(),
+                            context.getTagName(),
+                            context.getDataType(),
+                            context.getUserProperties()));
+                    break;
+                case ALREADY_PRESENT:
+                    final String newTagName = legacyEipAdapterConfig.getId() + "-" + UUID.randomUUID().toString();
+                    log.warn(
+                            "While migrating the EIPConfig a tag could not be added because a tag with the same name '{}' was already present. Another tagName using an random Uuid is used instead: '{}'",
```

Comment:
Since we are also trying to make things accessible via UI I think this should somehow be communicated to the UI.
At the very least I'd recommend to issue an Event.

Reply:
yeah it should create an event